File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max_val = 0 , min_val=0;
        int left = 0;
        int right = height.length -1;

        while(left<right){
            int dis = right-left;
            if (height[left] < height[right])  min_val = height[left];
            else min_val = height[right];

            int area = dis* min_val;

            if(max_val < area) max_val = area;

            if (height[left] < height[right]) left++;
            else right --;
            
        }
        return max_val;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        max_val = 0
        left = 0
        right = len(height)-1

        while left<=right:

            dis = right-left
            min_val = min(height[left],height[right])
            max_val = max(max_val, dis*min_val)
      
            if height[left]<height[right]:
                left+=1
            
            else:
                right-=1
        return max_val